src/DotNetEnv/Parsers.cs

```diff
@@ -174,16 +177,23 @@ private static string ToUtf32Char (string hex)
                 .Or(EscapedChar);
 
         // unquoted values can have interpolated variables,
-        // but only inline whitespace -- until a comment,
-        // and no escaped chars, nor byte code chars
-        // FIXME: would be nice to solve multi word with comment case with parser directly (no split + trim)
+        // but only inline whitespace, and no quote chars, no escaped chars, nor byte code chars
         internal static readonly Parser<ValueCalculator> UnquotedValue =
             InterpolatedValue.Or(
-                NotControlNorWhitespace("'\"$")
-                .Or(InlineWhitespaceChars.AtLeastOnce().Text())
-                .AtLeastOnce()
-                .Select(strs => new ValueActual(strs))
-            ).Many().Select(vs => new ValueCalculator(vs).Split("[ \t]#").Trim());
+                InlineWhitespaceChars.Many().Then(w =>
```

Comment:
My perspective on my implementation, to answer my own question from https://github.com/tonerdo/dotnet-env/pull/82/files#r1432311810 -- I like my approach better because
1. no regex
2. this is more straightforward to me: it is an interpolated value, or any inline whitespace followed by one or more acceptable printable chars (except a leading `#`), as many times as we can